nuxeo/templates/deployment.yaml

```diff
@@ -1,24 +1,36 @@
+{{- range splitList "," (include "nuxeo.nodeTypes" .) }}
+{{- $nuxeoNodeType := . -}}
+{{- $nuxeoFullnameSuffix := ternary "" (printf "-%s" $nuxeoNodeType) (eq $nuxeoNodeType "single") -}}
+{{- $workerNode := (eq $nuxeoNodeType "worker") -}}
+{{- with $ }}
+---
 apiVersion: apps/v1
 kind: Deployment
 metadata:
-  name: {{ template "nuxeo.fullname" . }}
+  name: {{ template "nuxeo.fullname" . }}{{ $nuxeoFullnameSuffix }}
   labels:
     chart: "{{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}"
 spec:
   selector:
     matchLabels:
       app: {{ template "nuxeo.fullname" . }}
-  replicas: {{ .Values.replicaCount }}
+      nuxeoNode: {{ $nuxeoNodeType }}
+  replicas: {{ ternary (default 1 .Values.workerCount) .Values.replicaCount $workerNode }}
   template:
     metadata:
       labels:
         app: {{ template "nuxeo.fullname" . }}
+        nuxeoNode: {{ $nuxeoNodeType }}
+        tier: {{ ternary "backend" "frontend" $workerNode }}
```

Comment:
I'm wondering if it is interesting to have these labels on every created resource?

Reply:
Right, thanks. I'll add them to the ConfigMap since it can be duplicated for the api and worker case.